### PR TITLE
Support page

### PR DESCRIPTION
**Pull request by JanPetr**

### CODE REVIEW

File: view/adminhtml/templates/support/contact.phtml [hunk 1]
```diff
@@ -0,0 +1,144 @@
+<?php /** @var $this \Algolia\AlgoliaSearch\Block\Adminhtml\Support\Contact */
+
+use Magento\Backend\Block\Template;
+
+?>
+
+<?php echo $block
+    ->getLayout()
```
Comment (damcou) [inline view/adminhtml/templates/support/contact.phtml]: Should be better in a block method

File: view/adminhtml/templates/support/contact.phtml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    <div class="admin__field-note">
+                        <div class="note_icon">i</div>
+                        <span>
+                            To provide a better support, this information will be attached to your request:
+                            <ul>
+                                <li>Magento and Algolia extension version numbers</li>
+                                <li>Algolia extension configuration</li>
+                                <li>
+                                    Indexing queue information
+                                    <?php echo $block
```
Comment (damcou) [inline view/adminhtml/templates/support/contact.phtml]: Should be better in a block method

File: view/adminhtml/templates/support/contact.phtml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                            type="checkbox"
+                            class="admin__actions-switch-checkbox"
+                            id="send_additional_info"
+                            name="send_additional_info"
+                            value="1"
+                            checked />
+                        <label class="admin__actions-switch-label" for="send_additional_info">
+                            <span class="admin__actions-switch-text">
+                                Attach list of installed 3rd party modules and catalog size
+                                <?php echo $block
```
Comment (damcou) [inline view/adminhtml/templates/support/contact.phtml]: Should be better in a block method

File: view/adminhtml/templates/support/overview.phtml [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+$noResultsIllustration = $this->getViewFileUrl('Algolia_AlgoliaSearch::images/support-no-results.svg');
+
+$videoDiscover = $this->getViewFileUrl('Algolia_AlgoliaSearch::images/video-discover.png');
+$videoQueue = $this->getViewFileUrl('Algolia_AlgoliaSearch::images/video-queue.png');
+$videoFeatures = $this->getViewFileUrl('Algolia_AlgoliaSearch::images/video-features.png');
+
+?>
+
+<?php echo $block
+    ->getLayout()
```
Comment (damcou) [inline view/adminhtml/templates/support/overview.phtml]: Should be better in a block method

File: etc/adminhtml/menu.xml [hunk 5]
```diff
@@ -54,5 +54,14 @@
             action="algolia_algoliasearch/reindex/index"
             resource="Algolia_AlgoliaSearch::manage"
         />
+        <add
+            id="Algolia_AlgoliaSearch::support"
+            title="Help &amp; Support"
+            module="Algolia_AlgoliaSearch"
+            sortOrder="120"
```
[the remark this answers is not in the document]
Comment (JanPetr) [reply etc/adminhtml/menu.xml]: I wanted to display it at the bottom now as we don't know what comes in a future. When we have something else what we want to display in front of it, we can always update the `sortOrder` with it. No?

File: view/adminhtml/templates/support/overview.phtml [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+            <div class="main">
+                Documentation
+            </div>
+
+            <div class="sub">
+                Documentation for all things Algolia for Magento
+            </div>
+        </div>
+        <div class="links">
+            <a href="https://community.algolia.com/magento/doc/m1/indexing/?utm_source=magento&utm_medium=extension&utm_campaign=support-page" target="_blank">
```
[the remark this answers is not in the document]
Comment (JanPetr) [reply view/adminhtml/templates/support/overview.phtml]: Great catch!

File: view/adminhtml/templates/support/contact.phtml [hunk 7]
```diff
@@ -0,0 +1,144 @@
+<?php /** @var $this \Algolia\AlgoliaSearch\Block\Adminhtml\Support\Contact */
+
+use Magento\Backend\Block\Template;
+
+?>
+
+<?php echo $block
```
[the remark this answers is not in the document]
Comment (JanPetr) [reply view/adminhtml/templates/support/contact.phtml]: Definitely, refactored

File: view/adminhtml/templates/support/contact.phtml (hunk 1 above)
Comment (JanPetr) [reply view/adminhtml/templates/support/contact.phtml]: 👍

File: Block/Adminhtml/Support/Contact.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+namespace Algolia\AlgoliaSearch\Block\Adminhtml\Support;
+
+use Algolia\AlgoliaSearch\Helper\SupportHelper;
+use Magento\Backend\Block\Template;
+use Magento\Backend\Block\Template\Context;
+use Magento\Backend\Model\Auth\Session;
+use Magento\Framework\Module\ModuleListInterface;
+
+class Contact extends Template
```
[the remark this answers is not in the document]
Comment (JanPetr) [reply Block/Adminhtml/Support/Contact.php]: Great catch! Thanks for posting that, I'm going through it and see what we can do about that. The Template class file mentions the same: https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Backend/Block/Template.php#L15-L27

File: Controller/Adminhtml/Support/Contact.php [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+            $this->messageManager->addErrorMessage('Your Algolia app is not eligible for e-mail support.');
+
+            $resultRedirect = $this->resultRedirectFactory->create();
+            $resultRedirect->setPath('*/*/index');
+
+            return $resultRedirect;
+        }
+
+        $params = $this->getRequest()->getParams();
+        if ($this->isFormSubmitted($params) && $data = $this->validateForm($params)) {
```
[the remark this answers is not in the document]
Comment (JanPetr) [reply Controller/Adminhtml/Support/Contact.php]: I had that, but when the validation failed, I redirected back to Controller which served to form and I lost the filled in values in the form.
Do you know how to keep them without putting them to URL as GET parameters, @damcou?

File: Helper/SupportHelper.php [hunk 10]
```diff
@@ -0,0 +1,344 @@
...
+
+namespace Algolia\AlgoliaSearch\Helper;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+class SupportHelper
+{
+    const INTERNAL_API_PROXY_URL = 'https://magento-proxy.algolia.com/';
```
[the remark this answers is not in the document]
Comment (JanPetr) [reply Helper/SupportHelper.php]: Good catch! 👍

File: view/adminhtml/templates/support/contact.phtml (hunk 7 above)
[the remark this answers is not in the document]
Comment (JanPetr) [reply view/adminhtml/templates/support/contact.phtml]: Good catch! Resolved.

File: view/adminhtml/templates/support/contact.phtml (hunk 1 above)
Comment (JanPetr) [reply view/adminhtml/templates/support/contact.phtml]: 👍

File: Block/Adminhtml/Support/Contact.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (JanPetr) [reply Block/Adminhtml/Support/Contact.php]: Sorry, Friday 🤦‍♂️ I just noticed I posted the very same link as you did. 🤐

File: Controller/Adminhtml/Support/Contact.php (hunk 9 above)
[the remark this answers is not in the document]
Comment (bsuravech) [reply Controller/Adminhtml/Support/Contact.php]: I agree with @damcou and having the post action be separated from the view controller. 

You could probably store the form values in session if you wanted to preserve form values on invalidation redirect. You'd pull the default data from session if it's been applied.

**Pull request merged**